mlir/lib/Dialect/LLVMIR/IR/LLVMDialect.cpp

```diff
@@ -3076,6 +3076,36 @@ void InlineAsmOp::getEffects(
   }
 }
 
+//===----------------------------------------------------------------------===//
+// masked_gather (intrinsic)
+//===----------------------------------------------------------------------===//
+
+LogicalResult LLVM::masked_gather::verify() {
+  auto ptrsVectorType = getPtrs().getType();
+  Type expectedPtrsVectorType =
```

Comment:
nit: a comment about the specific issue we want to verify here would help understanding